### ISSUE TITLE
@Context

### ISSUE THREAD

**Issue by nadalin** [w3c/vc-data-model#604]
"Though this specification requires that a @context property be present, there is no requirement that the value of the @context property be processed using JSON-LD. This is to support processing using plain JSON libraries such as those that may be used when the verifiable credential is encoded as a JWT. JSON-LD aware processors can process the @context using full JSON-LD processing as expected. For other processors, the only processing necessary is to ensure that the order of the values in @context is what is expected for the particular application, but no JSON-LD processing of those values is required."

This is odd wording, as you are saying that "processing of the @Context is required" no matter if you are using JSON or JSON-LD, which is not stated in the @Context section at all, just in the note

**Comment by brentzundel** [comment]
I'm not sure what issue you are bringing up with this comment. What you have said is true, but is not clear to me what you feel ought to be changed.

**Comment by nadalin** [comment]
@brentzundel Issue as that can't even begin to parse what is needed to be processed when and by what

**Comment by chaals** [comment]
> the only processing necessary is to ensure that the order of the values in @context is what is expected for the particular application

This appears to be the key requirement for everyone.

> JSON-LD aware processors can process the @context using full JSON-LD processing as expected

That seems pretty clear.

The rest seems to be rationale.

Would it be clearer if we broke the paragraph into pieces? Or have I misinterpreted it too?

**Comment by TallTed** [comment]
A rephrasing is definitely worthwhile here. Something along the lines of the following might replace the entire quote in the OP here.  

> Plain JSON libraries or processors (such as those that may be used when the verifiable credential is encoded as a JWT) MUST ensure that the order of the values in the **@context** property is what is expected for the particular application. Libraries or processors that support JSON-LD SHOULD process the **@context** property using full JSON-LD processing.

**Comment by brentzundel** [comment]
Would it be accurate to clarify it even further?

All libraries or processors MUST ensure that the order of the values
in the *@context
<https://github.com/context>* property is what is expected for the
particular application. Libraries or processors that support JSON-LD SHOULD
process the *@context <https://github.com/context>* property using full
JSON-LD processing.

**Comment by msporny** [comment]
Adopt suggestions from @brentzundel to address this issue.

**Comment by msporny** [comment]
Non-substantive suggestions by @brentzundel have been merged. Suggest closing this issue since it's been addressed by PR #630.

**Comment by burnburn** [comment]
WG resolution on 28 May 2019 call:

RESOLUTION: Issue #604 is addressed via PR #630, which adds non-normative text to clarify the processing requirements around @context. Multiple other PRs were also made to address this issue. Issue #604 will be closed after a 7 day wait period.
